From de5db802074fbb5cfd1fc216fc43836899da5b8b Mon Sep 17 00:00:00 2001
From: hepengpx <pengpengx.he@intel.com>
Date: Mon, 6 Jul 2026 14:07:41 +0800
Subject: [PATCH 26/28] media: ipu6: clean csi2 fwnode ep when destroying v4l2
 async connection

Release the CSI-2 endpoint fwnode retained by the async connection when
the V4L2 async notifier destroys the connection.

Signed-off-by: He, Pengpeng <pengpengx.he@intel.com>
Signed-off-by: Yew, Chang Ching <chang.ching.yew@intel.com>
---
 drivers/media/pci/intel/ipu6/ipu6-isys.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/media/pci/intel/ipu6/ipu6-isys.c b/drivers/media/pci/intel/ipu6/ipu6-isys.c
index 7ddc47bb0c..6d13c19064 100644
--- a/drivers/media/pci/intel/ipu6/ipu6-isys.c
+++ b/drivers/media/pci/intel/ipu6/ipu6-isys.c
@@ -1033,9 +1033,18 @@ static int isys_notifier_complete(struct v4l2_async_notifier *notifier)
 	return v4l2_device_register_subdev_nodes(&isys->v4l2_dev);
 }
 
+static void isys_notifier_destroy(struct v4l2_async_connection *asc)
+{
+	struct sensor_async_sd *s_asd =
+		container_of(asc, struct sensor_async_sd, asc);
+
+	fwnode_handle_put(s_asd->csi2.ep);
+}
+
 static const struct v4l2_async_notifier_operations isys_async_ops = {
 	.bound = isys_notifier_bound,
 	.complete = isys_notifier_complete,
+	.destroy = isys_notifier_destroy,
 };
 
 #define ISYS_MAX_PORTS 8
-- 
2.53.0

